proposals/NNNN-migration-tooling-for-upcoming-features.md

```diff
@@ -0,0 +1,124 @@
+# Migration tooling for Swift upcoming features
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Implementation: TBD
+* Review: TBD
+
+## Introduction
+
+As Swift evolves, source-breaking changes to the language are staged behind new
+language modes and, more recently, [upcoming features][SE-0362] that may be
+enabled piecemeal without adopting an entire new language mode. 
+
+In this proposal, we zoom in on the experience of adopting *individual* 
+source-breaking upcoming features. The premise is that, in principle, migration 
+is generally separable from adoption (?), and that decoupling migration will 
+empower compiler engineers to offer significantly richer diagnostics and 
+guidance that would not be appropriate in regular builds.
+
+[SE-0362]: https://github.com/swiftlang/swift-evolution/blob/main/proposals/0362-piecemeal-future-features.md
+
+## Motivation
+
+Adjusting to new behaviors or language requirements may demand research, 
+collaboration across multiple teams, careful consideration, and manual code 
+refactorings, sometimes on a case-by-case basis. 
+For many projects, source-breaking language features are expected to generate 
+hundreds of targeted errors.
+It is also likely for developers to encounter knock-on collateral errors that
+are not directly indicative of the exact cause or source of the problem.
+We should strive for a better migration experience than blocking development on
+atomic adoption and requiring programmers to slog through such a volume of 
+diagnostics. 
+
+The general convention around compiler warnings is that they must provide a way
+to be silenced. Migration guidance 
+
+Code migrations can require time and, more importantly, collaboration across
+multiple teams 
+
+Something about the following being good illustrations of the problems 
+described above:
```

Comment:
For this section, I recommend going through all of the existing upcoming features and noting which ones are possible to automatically provide fix-its for in a migration mode (e.g. by inserting `any` or `some` for `ExistentialAny`, and making `public import`s explicit for `InternalImportsByDefault`)